### ISSUE TITLE
Terraform triggering AWS AttachGroupPolicy event even when relation already exists

### ISSUE THREAD

**Issue by bbdimitriu** [hashicorp/terraform#10946]
### Terraform Version
Terraform v0.8.2

### Affected Resource(s)
- aws_iam_group_policy_attachment

### Terraform Configuration Files
```hcl
resource "aws_iam_group" "mygroup" {
  name = "mygroup"
  path = "/"
}

resource "aws_iam_group_policy_attachment" "MyAttachment" {
  group = "${aws_iam_group.mygroup.name}"
  policy_arn = "${aws_iam_policy.mypolicy.arn}"
}

resource "aws_iam_policy" "mypolicy" {
  name = "mypolicy"
  policy = "${data.aws_iam_policy_document.mypolicy.json}"
}

data "aws_iam_policy_document" "mypolicy" {
  statement {
    effect = "Allow"
    actions = [
      "s3:ListBucket",
      "s3:PutObject",
    ]
    resources = [
      "arn:aws:s3:::*"
    ]
  }
}
```
### Expected Behavior
Both group `mygroup` and policy `mypolicy` already exist on AWS and are synced up with the state file and the configuration. Also the association between the group and the policy is already defined in the AWS console.
I'd expect that either there is way to import a `aws_iam_group_policy_attachment` resource, or Terraform would detect the existing association between the group and the policy and it would not trigger a AttachGroupPolicy API call (as observed in CloudTrail).

### Actual Behavior
`terraform plan` suggests a new resource of type `aws_iam_group_policy_attachment` should be created and when `terraform apply` is called it links the two entities, even though they are already linked though a policy attachment. No logical change takes place though.

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
None

### References
Possibly related to GH-9643

**Comment by grubernaut** [comment]
Hi @bbdimitriu, thanks for the issue!

After the initial change experienced from running a `terraform apply`, do subsequent runs of `terraform apply` still show the diff? 

There currently isn't a way to import an `aws_iam_group_policy_attachment`, although that's something that we should have. 

Were there any specific steps that you followed that brought you into the current state? I'm unable to reproduce the error based on running `terraform apply` on the configuration provided above, wondering if there were any additional steps to take for a full reproduction.
Thanks!

**Comment by bbdimitriu** [comment]
After the initial "terraform apply" the diff does not appear any more (so I guess that the state file is updated with the relation).
I can't think of anything specific that I may have done to get it into this state, although I would not exclude it.

**Comment by grubernaut** [comment]
That's interesting, were there any manual imports or manual state changes between the initial `apply` and the current state? Was this during a migration from a non-terraform managed infrastructure to a managed terraform infrastructure?

**Comment by bbdimitriu** [comment]
Sorry for the delayed response. Yes, this is a migration from a non-terraform managed infrastructure to a managed one. So I did manual imports of the group and the policy document resources. I could not import the policy attachment as well, as mentioned above.

**Comment by grubernaut** [comment]
Hi @bbdimitriu, thanks for the response!

Because Terraform does not automatically import resources, the diff that you saw on the initial `terraform plan` was technically correct. Once the resource was added to the statefile, the diff was no longer present. Adding `import` functionality for the `aws_iam_group_policy_attachment` would have solved this issue for you. 

Are you still experiencing a diff on `terraform plan` or has the problem been solved after the initial `terraform apply`? If the problem is no longer occurring, I can take this as a feature request for adding `import` for the `aws_iam_group_policy_attachment` resource. Thanks!

**Comment by bbdimitriu** [comment]
Let me try to explain one more time the steps in turn:

1. First, AWS Console has group, policy and policy attachment defined. None of these are in Terraform yet.
2. Then I "terraform import" the group and the policy document. Now my state file contains the group and the policy
3. Then I modify my Terraform file to define the group and policy document. Running "terraform plan" now shows no change.
4. Then I modify my Terraform file to define the `aws_iam_group_policy_attachment` and run "terraform plan". This shows me that it's about to `apply` the attachment, which is correct with regards to the state (even though the attachment already exists in AWS)
5. Then I run "terraform apply" and the policy attachment is created. At this point the existing policy attachment in AWS is probably overwritten (by the same logical entity). If I run "terraform plan" now I don't see and diff (which is also expected behaviour).

In conclusion this whole process makes the state file, Terraform config file and AWS all be in sync as expected, but it caused a mutable operation, which shouldn't be necessary: the overwriting of the already existing AWS policy attachment.
I guess this can effectively be seen as a feature request for adding `import` for the `aws_iam_group_policy_attachment` resource.

Thanks!
Bogdan


 _after_ I imported the group and policy document and _after_ I manually created an attachment policy (because it couldn't be imported). So my state file had the group + policy but didn't the attachment and my config file had group + policy + attachment.

**Comment by grubernaut** [comment]
@bbdimitriu that sounds correct to me! 

Yeah, if the `aws_iam_group_policy_attachment` resource allowed for an `import`, then the mutable operation would not have been a requirement to sync the Terraform statefile, config file, and AWS.

It is, however, possible to manually edit the statefile and achieve the same result as `terraform import`, but it carries a high amount of risk. I'll be looking at adding in an import function for the policy attachment resource as well. 
Thanks!
